src/cdk-experimental/menu/menu-bar.ts

```diff
@@ -20,18 +33,88 @@ import {CDK_MENU, Menu} from './menu-interface';
   selector: '[cdkMenuBar]',
   exportAs: 'cdkMenuBar',
   host: {
+    '(keydown)': '_handleKeyEvent($event)',
+    '(focus)': 'focusFirstItem()',
     'role': 'menubar',
+    'tabindex': '0',
     '[attr.aria-orientation]': 'orientation',
   },
   providers: [
     {provide: CdkMenuGroup, useExisting: CdkMenuBar},
     {provide: CDK_MENU, useExisting: CdkMenuBar},
   ],
 })
-export class CdkMenuBar extends CdkMenuGroup implements Menu {
+export class CdkMenuBar extends CdkMenuGroup implements Menu, OnDestroy, AfterContentInit {
   /**
    * Sets the aria-orientation attribute and determines where menus will be opened.
    * Does not affect styling/layout.
    */
   @Input('cdkMenuBarOrientation') orientation: 'horizontal' | 'vertical' = 'horizontal';
+
+  /** Handles keyboard events for the MenuBar. */
+  private _menuKeyManager?: MenuKeyManager;
+
+  /** All child MenuItem elements nested in this MenuBar. */
+  @ContentChildren(CdkMenuItem, {descendants: true})
+  private readonly _allItems: QueryList<CdkMenuItem>;
+
+  constructor(
+    private readonly _dir: Directionality,
+    @Inject(TYPE_AHEAD_DEBOUNCE) private readonly _debounceInterval: number
+  ) {
+    super();
+  }
+
+  ngAfterContentInit() {
+    super.ngAfterContentInit();
+
+    this._menuKeyManager = new MenuKeyManager(
+      this._allItems,
+      this._dir,
+      this.orientation,
+      this._debounceInterval
+    );
+  }
+
+  /** Place focus on the first MenuItem in the menu. */
+  focusFirstItem() {
+    if (this._menuKeyManager) {
+      this._menuKeyManager.focusFirstItem();
+    }
+  }
+
+  /** Place focus on the last MenuItem in the menu. */
+  focusLastItem() {
+    if (this._menuKeyManager) {
+      this._menuKeyManager.focusLastItem();
+    }
+  }
+
+  /** Place focus on the given MenuItem in the menu. */
+  focusItem(child: CdkMenuItem) {
+    if (this._menuKeyManager) {
+      this._menuKeyManager.focusItem(child);
+    }
+  }
+
+  /** Get an emitter which emits bubbled-up keyboard events from the keyboard manager. */
+  _getBubbledKeyboardEvents(): Subject<KeyboardEvent> | null {
+    if (this._menuKeyManager) {
+      return this._menuKeyManager._bubbledEvents;
+    }
+    return null;
```

Comment:
```suggestion
    return this?._menuKeyManager?._bubbledEvents || null;
```
Though you could omit the `|| null` if `undefined` is also an acceptable return value